Closes #2130 Cheatsheet

manciniedoardo opened the pull request:
Closes #2130 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

manciniedoardo commented:
Links to cheatsheet will be broken until we merge into main once -  this is what the README would look like: 
![image](https://github.com/pharmaverse/admiral/assets/53403957/c16358e9-edf8-40b3-950b-7e376685783c)
If you want to check fopr yourself, just go into the branch and replace _/main/_ in the links to _/2130_cheatsheet/_

bms63 commented:
whoa that looks nice!!

manciniedoardo commented:
> Do we want this to live in this package or in admiraldiscovery? If we put in admiraldiscovery and link to it, then we can update it if needed rather than waiting for a release

I see your point, on the other hand the cheatsheet should be quite static on should only update if a release happens anyway. And i feel it's more accessible if featured in the main package.

bms63 commented:
My twos hesitations are first if we fix something/update/supersede and it is referenced in the cheat sheet then we also need to update the cheat sheet.  If we forget and do a release, then are hands are tied.  If we keep it apart from admiral package, then we can easily update it if we forget.    Second, I don't think packages store cheatsheets in their packages unless I am mistaken.  

We have admiraldiscovery set up - why not just put it there and link to it in our website??

I guess that is another point - that we have to keep an eye on the cheat sheet going forward when making updates to admiral

ddsjoberg commented:
> Many cheat sheets are stored in https://github.com/rstudio/cheatsheets. I am not sure what common practice is. @ddsjoberg , what do you think?

I think keeping the on the RStudio helps discoverability since useRs can browse all cheatsheets available. BUT, if we were to make an update in the future, the delay between submitted a PR to that repo and it being published on their site can be significant. I think the first cheatsheet I submitted there took just over a month to be merged in (not because of comments on the PR, just a delay in merging). That said, I would just submit one there regardless, knowing that there may be short periods when it's not the most recent version available if there is a review delay.

Regarding keeping the raw materials in admiral vs admiraldiscovery: I don't think it makes a difference. I would imagine we just link to the PDF at the raw location on GitHub, so making releases doesn't matter in this case: we can updated it whenever we want.

bms63 commented:
Okay.  I guess we are placing into the admiral package.

We should make a note in the PR checklist to review the cheatsheet to see if an update is needed, similar to checking templates and vignettes.  

@manciniedoardo if someone has to make an update for the cheatsheet and regenerate - is the phuse link noted somewhere?

bms63 commented:
> > @manciniedoardo if someone has to make an update for the cheatsheet and regenerate - is the phuse link noted somewhere?
> 
> Great shout, I added an item to the PR template. I wonder though if the phuse link will make the links check fail? 🤔

We can just put it in the `.lycheeignore` since only devs would ever go and update it and we can give access.  Do you mind fixing the other links for `admiraldev` those all have the `devel` removed now.

manciniedoardo commented:
> We can just put it in the `.lycheeignore` since only devs would ever go and update it and we can give access. Do you mind fixing the other links for `admiraldev` those all have the `devel` removed now.

Fixed links and added PHUSE link to .lycheeignore but the check is still failing, am I doing smth wrong or does this require merging in first?

bundfussr commented on .lycheeignore:
```diff
@@ -3,3 +3,6 @@ https://github.com/pharmaverse/admiral/blob/main/
 irongut/CodeCoverageSummary@v1.2.0
 https://packagemanager.rstudio.com/cran/__linux__/focal/latest
+https://raw.githubusercontent.com/pharmaverse/admiral/main/inst/cheatsheet/cheatsheet_image.png
+inst/admiral_cheatsheet.pdf
+inst/admiral_cheatsheet.pptx
```
Is this correct?

manciniedoardo replied:
Is there a better way to write this?

bundfussr replied:
Do we have any links to `inst/admiral_cheatsheet.pdf` or `inst/admiral_cheatsheet.pptx`?

In the affirmative, should they be changed to `inst/cheatsheet/admiral_cheatsheet.pdf`?

bundfussr reviewed (changes requested):
`.Rbuildignore` needs to be updated.

An update of the cheat sheet is required due to changes implemented for the upcoming release, e.g., some example calls need to be updated and superseded functions should be removed.

Should we do this at the end when all PRs are merged? Or should this be handled in the PR which require a change of the cheat sheet?

bundfussr reviewed (changes requested):
The highlighted quotes should be removed.
![image](https://github.com/pharmaverse/admiral/assets/80953585/30cdaef2-02d1-4046-9c6e-8a814a775bbb)
